src/main/features/response/tasks/whenWillYouPayTask.ts

```diff
@@ -40,9 +40,28 @@ export class WhenWillYouPayTask {
 
   private static statementOfMeansIsCompletedIfApplicable (responseDraft: ResponseDraft): boolean {
     if (StatementOfMeans.isApplicableFor(responseDraft)) {
-      return responseDraft.statementOfMeans !== undefined
-        && isValid(responseDraft.statementOfMeans.residence)
+      const statementOfMeans: StatementOfMeans = responseDraft.statementOfMeans
+
+      return statementOfMeans !== undefined
+        && isValid(statementOfMeans.residence)
+        && WhenWillYouPayTask.isDependantsCompleted(statementOfMeans)
+        && isValid(statementOfMeans.employment)
+        && isValid(statementOfMeans.bankAccounts)
     }
+
     return true
   }
+
+  private static isDependantsCompleted (statementOfMeans: StatementOfMeans): boolean {
```

Comment:
Any reason `isValid` cannot be used here? We should aim to have just one validation logic and reuse it when task list is evaluated. Usually when view model is invalid task is not completed.

Reply:
This logic is FAR more complex than just checking if something `isValid` as there is so many different flows that depends on what you've previously selected.

This is why I wanted to do it when all pages are provided. It's going to become even more complex as there will be 2 more pages just for dependants and employment

Reply:
You are right, there are some extras checks I did not notice staring away. Posted alternate implementation.